src/main/resources/controllers-context.xml

```diff
@@ -61,14 +61,14 @@
     </mvc:annotation-driven>
 
     <beans profile="transactionalDogService">
-        <bean id="transactionalDogService" class="com.example.dog.service.TransactionalDogService">
+        <bean id="proxy" class="com.example.dog.service.TransactionalDogService">
             <constructor-arg ref="jdbcDogService"/>
             <constructor-arg ref="jdbcConnectionHolder"/>
         </bean>
     </beans>
 
     <beans profile="transactionalProxy">
-        <bean id="transactionalProxy" class="com.example.dog.service.TransactionalProxy" factory-method="createProxy">
+        <bean id="proxy" class="com.example.dog.service.TransactionalProxy" factory-method="createProxy">
```

Comment:
Maybe at least transactionalDogService? :)

Reply:
Why `transactionalDogService`? Now we use `transactionalProxy` + should't it be something neutral?

Reply:
What would you call it when you create OwnerService and its transactional proxy?

Reply:
Well, fine, then...but for me it's a bit strange to put `transactionalDogService` into Controller's constructor, I'd think that I really use now `transactionalDogService` bean (but it may not be true), that's why I'm talking about absolutely neutral name

Reply:
In reality you wouldn't have 2 ways of creating this tx dogservice. We're just playing with profiles.